### PR TITLE
[Homework #9] - JS Advanced 2 By Pavel Didenko

### PR DESCRIPTION
**Pull request by Win4esterD**
I think I implemented all what is needed, except the point

> non-valid / non-expected content of serialized line must call for an exception in case of waking up``

Don't know what scenarios does it have to handle. I understand that a person can add something to a serialized object, or remove something from it, but have no idea of how to spot it and throw exception.

### CODE REVIEW

File: homeworks/pavel-didenko/09-js-advanced-2/task9.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      for(let item of resultKeys){
+        if (!constructorKeys.includes(item)) {
+          throw new Error(`Not instance of ${this.constructor.name}`);
+        }
+      }
+      return new this.constructor(result);
+    }
+  }
+
+  nAnStringifier(key, value){
```
Comment (ilya-ischenko) [inline homeworks/pavel-didenko/09-js-advanced-2/task9.js:26]: "key" argument is unused, you can mark it as "_",  -  nAnStringifier(_, value)

File: homeworks/pavel-didenko/09-js-advanced-2/task9.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+class Serializable {
+  static serializedData = {};
```
Comment (ilya-ischenko) [inline homeworks/pavel-didenko/09-js-advanced-2/task9.js:4]: unused

**Review by ilya-ischenko — COMMENTED** [review]
1. Infinity value lost after serialization, it becomes string after wakeFrom method
2. Need to add some validation - this strings throw enexpected errors:
new UserDTO().wakeFrom([]);
new UserDTO().wakeFrom({});

**Comment by Win4esterD** [comment]
> 1) Infinity value lost after serialization, it becomes string after wakeFrom method

Resolved. It's strange that it revived "Infinities" for the instances of Singer class, but didn't work for UserDTO

> 2. Need to add some validation - this strings throw enexpected errors:
>    new UserDTO().wakeFrom([]);
>    new UserDTO().wakeFrom({});

Since it's supposed to accept type string as the parameter, I just made it o throw an error.

image

File: homeworks/pavel-didenko/09-js-advanced-2/task9.js (hunk 1 above)
Comment (Win4esterD) [reply homeworks/pavel-didenko/09-js-advanced-2/task9.js]: Resolved, changed "key" to "_" in lines 28 and 39.

File: homeworks/pavel-didenko/09-js-advanced-2/task9.js (hunk 2 above)
Comment (Win4esterD) [reply homeworks/pavel-didenko/09-js-advanced-2/task9.js]: Ups, it left from my previous solution where I stored values for future deserialization in static property.

Removed.

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**